source/common/http/conn_manager_impl.cc

```diff
@@ -1173,45 +1173,66 @@ void ConnectionManagerImpl::ActiveStream::refreshDurationTimeout() {
   }
   auto& route = filter_manager_.streamInfo().route_entry_;
 
-  auto grpc_timeout = Grpc::Common::getGrpcTimeout(*request_headers_);
   std::chrono::milliseconds timeout;
   bool disable_timer = false;
+  auto grpc_timeout = Grpc::Common::getGrpcTimeout(*request_headers_);
```

Comment:
Offhand I recall debate around the original PR about how the new fields should be applied and it was the consensus from the gRPC folks that the difference wasn't "is this a gRPC request or not" but "is the gRPC timeout header present or not" so I think switching to this is against the intent of the folks who added this to the API.

Reply:
Got it. This is good to know. I will change it accordingly

Reply:
Awesome.  It's super complicated and confusing, so thanks for continuing to plug away at it!